crates/tabby/src/main.rs

```diff
@@ -54,19 +56,35 @@ pub enum Commands {
     /// Execute the repository sync job.
     #[cfg(feature = "ee")]
     #[clap(name = "job::sync")]
-    JobSync,
+    JobSync(JobArgs),
 
     /// Execute the index job.
     #[cfg(feature = "ee")]
     #[clap(name = "job::index")]
-    JobIndex,
+    JobIndex(JobArgs),
+}
+
+#[derive(clap::Args)]
+pub struct JobArgs {
+    #[clap(long, requires = "address")]
+    token: Option<String>,
+    #[clap(long, requires = "port")]
+    addr: Option<String>,
+    #[clap(long, requires = "token")]
+    port: Option<u16>,
 }
 
 #[derive(clap::Args)]
 pub struct SchedulerArgs {
     /// If true, runs scheduler jobs immediately.
     #[clap(long, default_value_t = false)]
     now: bool,
+    #[clap(long, requires = "address")]
+    token: Option<String>,
+    #[clap(long, requires = "port")]
+    addr: Option<String>,
+    #[clap(long, requires = "token")]
+    port: Option<u16>,
```

Comment:
When running with scheduler, there's no need to pass this three args (as scheduler runs fully local, no distribution support)